.travis.yml

```diff
@@ -22,9 +22,10 @@ before_install:
   - mysql -e 'CREATE DATABASE IF NOT EXISTS test;'
   - sudo mysql -e "use mysql; update user set authentication_string=PASSWORD('root') where User='root'; update user set plugin='mysql_native_password';FLUSH PRIVILEGES;"
   - sudo apt-get -y install libvirt-dev
+  - pip install pipenv
 
 install:
-  - pip install -r requirements.txt
```

Comment:
Here's a small observation I'd want you to make. So, on a normal system (not Travis), you'd want to change the script command to `pipenv run coverage ...` because pipenv install creates a new virtual env and installs the dependencies inside it. It will NOT work if the particular env shell is not invoked. Here on travis we are already inside a virtual env and hence no new env is created and the installation is done in the same.

Also, since @saurabhshri has suggested to keep the requirements file as well (which I'm not in support of since both will have to be maintained separately), it is wise to not make this change. So, say I install a dependency using pipenv inside the environment and forget to update requirements.txt manually (very likely), the tests will most probably fail and I would be reminded to update the file. Otherwise, I'm pretty sure, requirements.txt will get outdated pretty soon.

Reply:
So codecov have to be included in Pipfile, I'll put it on dev-packages in Pipfile and run `pipenv install --dev` instead, then run `pipenv run coverage ...`. How to test if the requirements.txt was outdated? Because the installation will be done in the same way, we won't know it.